File: task-bazel/AbstractFactory/cpp-source/rooms/BasementNormalRoom.cpp
//
// Created by Pavel Akhtyamov on 2019-02-25.
//

#include "BasementNormalRoom.h"

std::string BasementNormalRoom::GenerateBoss() const {
  return "100 hp";
}

std::string BasementNormalRoom::GenerateLoot() const {
  return "Loot";
}

size_t BasementNormalRoom::chance() const {
  return 1;
}

File: JavaSource/src/main/java/ru/akhcheck/patterns/abstractfactory/rooms/BasementNormalRoom.java
package ru.akhcheck.patterns.abstractfactory.rooms;

/** Обычная комната подвала. */
public class BasementNormalRoom implements NormalRoom {
    @Override
    public String generateLoot() {
        return "Basement loot";
    }

    @Override
    public String generateBoss() {
        return "Monstro";
    }
}
